### PR TITLE
supervisord: send more appropriate signals to daemons 

### PR DESCRIPTION
**Pull request by risicle**
I _think_ this is it.

https://trello.com/c/skD666lL

Use a combination of signals that are more likely to allow requests to complete. this is an attempt to reduce or eliminate unpleasant 502s being sent to users due to requests being aborted while they're still in flight when the instance is terminated for any reason.

The awscli cwlogs daemon doesn't appear to perform any explicit form of signal handling, so we might need to tailor this further if we feel it needs any more encouragement to send out any remaining logs before it is killed.

I've managed to terminate busy instances with this setup resulting in no 502s.

### CODE REVIEW

**Comment by agalamatis** [comment]
From what I read thus far, the only difference between TERM and INT is that the later can be sent with ctrl+C. But uswgi and the rest of the programs in the bash scripts might treat them differently.

File: supervisord.conf [hunk 1]
```diff
@@ -3,33 +3,34 @@ nodaemon = true
...
 stdout_logfile_backups = 3
 stderr_logfile = /dev/stderr
 stderr_logfile_maxbytes = 0
-stopsignal = INT
 
 [program:uwsgi]
-command = uwsgi --ini /etc/uwsgi.conf --master
+command = uwsgi --ini /etc/uwsgi.conf --hook-master-start "unix_signal:1 gracefully_kill_them_all" --master
+# see https://github.com/unbit/uwsgi/issues/849#issuecomment-118869386 in regards to signal handling
+stopsignal = HUP
```
Comment (lfdebrux) [inline supervisord.conf]: If we're using HUP does that mean we no longer have the ability to reread config? Can we use the SIGINT signal number instead?

Comment (lfdebrux) [reply supervisord.conf]: https://uwsgi-docs.readthedocs.io/en/latest/Management.html#signals-for-controlling-uwsgi suggests that normally SIGTERM and SIGINT do the same thing, so mapping graceful shutdown to SIGINT doesn't lose us anything

Comment (risicle) [reply supervisord.conf]: Sure.

**Comment by risicle** [comment]
See also https://trello.com/c/PyDaRXEj/

File: supervisord.conf (hunk 1 above)
Comment (risicle) [reply supervisord.conf]: Done.

**Pull request merged**